### PR TITLE
Refactor some of the root files, that are there for building

### LINKED ISSUE #221: Refactor Idea
To move the `docker-compose.*.yml` templates used with `make docker-compose.yml` to a sub directory.

Doing this would clean up the root directory and have the singular `docker-compose.yml` that is compiled + a system `docker-compose.override.yml`

### PR DESCRIPTION
**Pull request by misilot**
Refactor docker-compose files and demo-data to a build directory

This cleans up the root directory, with files that are utilized for building `isle-dc` but aren't used once the system is up and running, since `docker-compose.yml` is generated from all the smaller config files.

Resolves #221 

I can squash any of the commits / remove them. Sorry if I got a little over zealous.

### CODE REVIEW

**Comment by kstapelfeldt** [comment]
@DonRichards are you available to review this?

**Comment by misilot** [comment]
~~I did notice that when enabling the custom environment~~

~~the build context for drupal gets modifed with the `make -B docker-compose.yml`~~

```yml
  drupal:
    build:
      context: /Users/tmisilo/isle-dc/build/docker-compose
```
~~when it should just be `/Users/tmisilo/isle-dc` or even `./`~~

I fixed this in the latest commit with `context: ../../`

File: build/docker-compose/docker-compose.traefik.yml [hunk 1]
```diff
@@ -51,7 +51,7 @@ services:
       - ${CODE_SERVER_PORT:-8443}:8443   # code-server
     volumes:
       - /var/run/docker.sock:/var/run/docker.sock
-      - ./tls.yml:/etc/traefik/tls.yml
+      - ./build/traefik-tls.yml:/etc/traefik/tls.yml
```
Comment (DonRichards) [inline build/docker-compose/docker-compose.traefik.yml]: I think this might be causing a problem.

**Comment by DonRichards** [comment]
I'm getting a "ERR_CERT_AUTHORITY_INVALID". This is something Firefox will ignore but not chrome.

**Comment by DonRichards** [comment]
I see some stuff that gives a clue to the issue. The **build/docker-compose/build/certs** directory is empty. The a new folder "build/docker-compose/build/traefik-tls.yml/". The file build/docker-compose/docker-compose.traefik.yml is pointing to a relative file but the "build/" parent directory was added. I think dropping this might correct the issue.

File: sample.Dockerfile [hunk 2]
```diff
@@ -1,15 +1,20 @@
 # syntax=docker/dockerfile:experimental
 ARG REPOSITORY=islandora
-ARG TAG=latest
-FROM ${REPOSITORY}/drupal:${TAG}
+ARG TAG=ff1c469cce4227755298b7c98144b4af697363b4
```
Comment (DonRichards) [inline sample.Dockerfile]: Why was this specified?

**Comment by misilot** [comment]
@DonRichards I believe I fixed this, if you want to take a look.

Thanks!

File: sample.Dockerfile (hunk 2 above)
Comment (DonRichards) [reply sample.Dockerfile]: OK, no need to have this hold up the merge.

**Review by DonRichards — DISMISSED** [review]
Tested this

**Comment by DonRichards** [comment]
Because of the nature of this I'm going to wait 24 hours for feedback before I merge.

File: sample.Dockerfile (hunk 2 above)
Comment (misilot) [reply sample.Dockerfile]: I don't think I intended for this, other than `latest` is not a tag that is available on the containers, so I had to specify it. I will revert that change.

File: build/docker-compose/docker-compose.traefik.yml [hunk 3]
```diff
@@ -51,8 +51,8 @@ services:
       - ${CODE_SERVER_PORT:-8443}:8443   # code-server
     volumes:
       - /var/run/docker.sock:/var/run/docker.sock
-      - ./tls.yml:/etc/traefik/tls.yml
-      - ./certs:/etc/ssl/traefik
+      - ../../build/traefik-tls.yml:/etc/traefik/tls.yml
```
Comment (misilot) [inline build/docker-compose/docker-compose.traefik.yml:56]: @DonRichards I think this addresses what you might have been mentioning. Though I thought I had fixed this before.

File: build/docker-compose/docker-compose.traefik.yml (hunk 1 above)
Comment (misilot) [reply build/docker-compose/docker-compose.traefik.yml]: I think I had addressed that.  It should be ` ../../build/traefik-tls.yml:/etc/traefik/tls.yml`. I commented on the updated code

**Comment by misilot** [comment]
I rebased to the development branch.

**Comment by DonRichards** [comment]
Let me test again with that rebase and those changes

**Comment by misilot** [comment]
I just rebased with the latest changes in `development`

**Comment by misilot** [comment]
I just rebased everything with the latest merge :)

**Comment by DonRichards** [comment]
@misilot I hate to ask but would you rebase again? There's been no response from others so I'll test and merge it as soon as it's rebased.

**Comment by misilot** [comment]
@DonRichards rebased :) Thanks!

**Comment by misilot** [comment]
@DonRichards just checking in, if you ran into any issues.

Thanks!
Tom

**Comment by misilot** [comment]
rebased with latest changes

**Comment by misilot** [comment]
Rebased with latest changes.

**Comment by DonRichards** [comment]
@misilot I just checked this. It's introducing a new issue with generating secrets. check-secrets.sh didn't get updated with the "/build/" to the paths needed. Line #79 & #161. Also, the "find" command needs a leading "../" on line #28 example `find ../secrets/template/*` 
Once this is pushed I'll approve it to be merged.

**Comment by DonRichards** [comment]
Hoping this will help.
Screen Shot 2022-06-06 at 2 46 16 PM
Screen Shot 2022-06-06 at 2 46 05 PM

**Comment by misilot** [comment]
@DonRichards I believe I made the requested changes.  Thanks!

**Review by DonRichards — APPROVED** [review]
Tested, ready to merge

**Pull request merged**